### PR TITLE
Regenerate clients for ManagedInstanceProxyOverride

### PR DESCRIPTION
**Pull request by jamestao**

### CODE REVIEW

File: src/SDKs/SqlManagement/Management.Sql/Microsoft.Azure.Management.Sql.csproj [hunk 1]
```diff
@@ -7,7 +7,7 @@
 		<PackageId>Microsoft.Azure.Management.Sql</PackageId>
 		<Description>Azure SQL Management SDK library</Description>
 		<AssemblyName>Microsoft.Azure.Management.Sql</AssemblyName>
-		<Version>1.30.0-preview</Version>
+		<Version>1.31.0-preview</Version>
 		<PackageTags>Microsoft Azure SQL Management;SQL;SQL Management;</PackageTags>
 		<PackageReleaseNotes>
 			<![CDATA[
```
Comment (dsgouda) [inline src/SDKs/SqlManagement/Management.Sql/Microsoft.Azure.Management.Sql.csproj]: Please update the PackageReleaseNotes

**Review by dsgouda — CHANGES_REQUESTED** [review]

**Comment by dsgouda** [comment]
@jamestao Please link the REST spec PR

File: src/SDKs/SqlManagement/Management.Sql/Microsoft.Azure.Management.Sql.csproj (hunk 1 above)
Comment (jamestao) [reply src/SDKs/SqlManagement/Management.Sql/Microsoft.Azure.Management.Sql.csproj]: There's no update to the release notes, this reverted a breaking change that was discovered in 1.30.0-preview, and that version was delisted. Nothing new was added.

**Comment by jamestao** [comment]
specs PR: https://github.com/Azure/azure-rest-api-specs/pull/5780

**Comment by jaredmoo** [comment]
Can you in some way reference the enum class from unit tests so that build breaks if it's accidentally removed again?

**Comment by jamestao** [comment]
@jaredmoo In the interest of time would prefer to revert the change first, then ask the owner @stefanb995 to add proper tests for the enums.

**Review by jaredmoo — APPROVED** [review]

File: src/SDKs/SqlManagement/Sql.Tests/ManagedInstanceCrudScenarioTests.cs [hunk 2]
```diff
@@ -38,7 +38,7 @@ public void TestCreateUpdateGetDropManagedInstance()
                 string location = "westcentralus";
 
                 bool publicDataEndpointEnabled = true;
-                string proxyOverride = "Proxy";
+                string proxyOverride = ManagedInstanceProxyOverride.Proxy;
```
Comment (johnpaulkee) [inline src/SDKs/SqlManagement/Sql.Tests/ManagedInstanceCrudScenarioTests.cs]: nice

Comment (johnpaulkee) [reply src/SDKs/SqlManagement/Sql.Tests/ManagedInstanceCrudScenarioTests.cs]: this should help us catch this next time.

**Review by johnpaulkee — APPROVED** [review]

**Review by dsgouda — APPROVED** [review]
LGTM

**Pull request merged**